File: jwt/src/main/java/com/member/jwt/repository/RefreshRepository.java
package com.member.jwt.repository;

import com.member.jwt.entity.RefreshEntity;
import jakarta.transaction.Transactional;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface RefreshRepository extends JpaRepository<RefreshEntity, Long> {

    Boolean existsByRefresh(String refresh);

    // 이메일로 리프레시 토큰을 조회
    Optional<RefreshEntity> findByEmail(String email);

    // 이메일로 리프레시 토큰을 삭제
    @Transactional
    void deleteByEmail(String email);
}

File: jwtKotlin/src/main/kotlin/com/member/jwtkotlin/repository/RefreshRepository.kt
package com.member.jwt.repository

import com.member.jwtkotlin.entity.RefreshEntity
import jakarta.transaction.Transactional
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface RefreshRepository : JpaRepository<RefreshEntity?, Long?> {
    fun existsByRefresh(refresh: String?): Boolean?

    // 이메일로 리프레시 토큰을 조회
    fun findByEmail(email: String?): Optional<RefreshEntity?>?

    // 이메일로 리프레시 토큰을 삭제
    @Transactional
    fun deleteByEmail(email: String?)
}